providers/base/bin/switch_power_mode.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# All rights reserved.
+#
+# Written by:
+#   Bin Li <bin.li@canonical.com>
+
+"""Modules providing a function running os or sys commands."""
+import os
+import sys
+
+
+def main():
+    """Switching the power mode to check if the power mode can be switched."""
+    sysfs_root = "/sys/firmware/acpi/"
+    if not os.path.isdir(sysfs_root):
+        return
+    choices_filename = os.path.join(sysfs_root, "platform_profile_choices")
+    if (not os.path.isfile(choices_filename) or
+            not os.access(choices_filename, os.R_OK)):
+        return
+    profile_filename = os.path.join(sysfs_root, "platform_profile")
+    if (not os.path.isfile(profile_filename) or
+            not os.access(profile_filename, os.R_OK)):
+        return
+
+    return_value = 0
+
+    # Get the current power mode in quiet mode.
+    old_profile = os.popen("powerprofilesctl get").read().strip().split()[0]
+
+    # Read the power mode from /sys/firmware/acpi/platform_profile_choices
+    with open(choices_filename, "rt", encoding="utf-8") as stream:
+        choices = stream.read().strip().split()
+        if len(choices) < 1:
+            print("No power mode to switch.")
+            return
+    print(f'Power mode choices: {choices}')
```

Comment:
The checkbox codebase has to be compliant with python3.5.1 (sadly), so we can't use f-strings yet. Please change this to `print('Power mode choices: {}'.format(choices))`

Reply:
I saw there are more, please replace them all with a format call

Reply:
@Hook25 Thanks for your feedback.  I ran pylint to my code and found below error. Anyway, I will use .format to compliant with python3.5.1.

providers/base/bin/switch_power_mode.py:37:10: C0209: Formatting a regular string which could be a f-string (consider-using-f-string)

Reply:
pylint is correct, he can't know that you have to support python3.5